apps/wing/src/commands/init.ts

```diff
@@ -0,0 +1,167 @@
+import { existsSync, readdirSync } from "fs";
+import { copyFile, mkdir, readdir } from "fs/promises";
+import { join, relative } from "path";
+import chalk from "chalk";
+import inquirer from "inquirer";
+import { exists } from "./pack";
+
+const INIT_TEMPLATES_DIR = join(__dirname, "..", "..", "init-templates");
+
+/**
+ * Options for the `new` command.
+ * This is passed from Commander to the `init` function.
+ */
+export interface InitOptions {
+  readonly template?: string;
+  readonly language?: string;
+}
+
+/**
+ * Compiles a Wing program. Throws an error if compilation fails.
+ * @param entrypoint The program .w entrypoint.
+ * @param options Compile options.
+ * @returns the output directory
+ */
+export async function init(template: string, options: InitOptions = {}): Promise<void> {
+  const templates = initTemplateNames();
+  let language = options.language ?? "wing";
+
+  // If no template is specified, ask the user to specify one
+  if (!template) {
+    console.log("Usage: wing new <template> [--language <language>]");
+    console.log();
```

Comment:
Maybe we just enforce this in CLI argument by using choice?

Reply:
Here I was thinking to include an interactive picker so you can run `wing new` and see the all the templates available